mediagrains/utils/asyncbinaryio.py

```diff
@@ -378,10 +378,19 @@ async def truncate(self, size: Optional[int] = None) -> int:
         raise UnsupportedOperation("Cannot truncate a network stream")
 
     def tell(self) -> int:
-        return self._pos
+        if self._next_pos > self._pos:
```

Comment:
Nitpick: Can `self._next_pos` ever end up before `self.pos` (I assume not since you can't seek backwards)? In which case I think `return self_next_pos` always works here?
Unless you're aiming for the same code as in `seek`, in which case that makes sense

Reply:
The `self._next_pos` is not updated when data is read and so it will be before `self._pos` after a `read`. That's why `self._pos` needs to be returned in `tell` and used in `seek` as `self._next_pos` is not up-to-date.

My aim was to not re-write the code as it's probably a matter of style how to implement it. That means that `self._next_pos` is only a valid file position after a `seek` where it is guaranteed to be > `self._pos`. Otherwise `self._pos` should be used as the file position.